astropy/wcs/src/pyutil.c

```diff
@@ -372,591 +356,596 @@ wcshdr_err_to_python_exc(int status, const struct wcsprm *wcs) {
+                             propname, shape_str);
+                Py_DECREF(value_array);
+                return -1;
+            }
+        }
+    }
 
-  return 0;
-}
+    copy_array_to_c_int(value_array, dest);
 
-/* get_str_list is inlined */
+    Py_DECREF(value_array);
 
-int
-set_str_list(
-    const char* propname,
-    PyObject* value,
-    Py_ssize_t len,
-    Py_ssize_t maxlen,
-    char (*dest)[72]) {
-
-  PyObject*  str      = NULL;
-  Py_ssize_t input_len;
-  Py_ssize_t i        = 0;
-
-  if (check_delete(propname, value)) {
-    return -1;
-  }
-
-  if (maxlen == 0) {
-    maxlen = 68;
-  }
-
-  if (!PySequence_Check(value)) {
-    PyErr_Format(
-        PyExc_TypeError,
-        "'%s' must be a sequence of strings",
-        propname);
-    return -1;
-  }
-
-  if (PySequence_Size(value) != len) {
-    PyErr_Format(
-        PyExc_ValueError,
-        "len(%s) must be %u",
-        propname,
-        (unsigned int)len);
-    return -1;
-  }
-
-  /* We go through the list twice, once to verify that the list is
-     in the correct format, and then again to do the data copy.  This
-     way, we won't partially copy the contents and then throw an
-     exception. */
-  for (i = 0; i < len; ++i) {
-    str = PySequence_GetItem(value, i);
-    if (str == NULL) {
-      return -1;
-    }
-
-    if (!(PyBytes_CheckExact(str) || PyUnicode_CheckExact(str))) {
-      PyErr_Format(
-          PyExc_TypeError,
-          "'%s' must be a sequence of bytes or strings",
-          propname);
-      Py_DECREF(str);
-      return -1;
-    }
-
-    input_len = PySequence_Size(str);
-    if (input_len > maxlen) {
-      PyErr_Format(
-          PyExc_ValueError,
-          "Each entry in '%s' must be less than %u characters",
-          propname, (unsigned int)maxlen);
-      Py_DECREF(str);
-      return -1;
-    } else if (input_len == -1) {
-      Py_DECREF(str);
-      return -1;
-    }
-
-    Py_DECREF(str);
-  }
-
-  for (i = 0; i < len; ++i) {
-    str = PySequence_GetItem(value, i);
-    if (str == NULL) {
-      /* Theoretically, something has gone really wrong here, since
-         we've already verified the list. */
-      PyErr_Clear();
-      PyErr_Format(
-          PyExc_RuntimeError,
-          "Input values have changed underneath us.  Something is seriously wrong.");
-      return -1;
-    }
-
-    if (set_string(propname, str, dest[i], maxlen)) {
-      PyErr_Clear();
-      PyErr_Format(
-          PyExc_RuntimeError,
-          "Input values have changed underneath us.  Something is seriously wrong.");
-      Py_DECREF(str);
-      return -1;
-    }
-
-    Py_DECREF(str);
-  }
-
-  return 0;
+    return 0;
 }
 
+/* get_str_list is inlined */
 
-/*@null@*/ PyObject*
-get_pscards(
-    /*@unused@*/ const char* propname,
-    struct pscard* ps,
-    int nps) {
+int set_str_list(const char* propname, PyObject* value, Py_ssize_t len, Py_ssize_t maxlen,
+                 char (*dest)[72])
+{
```

Comment:
@pllim @mhvk since you have opinions, would you prefer the old format of one argument per line, black style?

Reply:
> one argument per line, black style?

Is saying "eww" against code of conduct?

Reply:
Let's not do one argument per line! (I.e., I like this.) The increase in line length and 4 space indentations seem to have solved most of my problems. I'm also terribly biased towards being close to numpy style, since that's most of the C I see...